src/main/java/dev/pgm/community/utils/WebUtils.java

```diff
@@ -10,20 +10,75 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.io.OutputStream;
 import java.net.HttpURLConnection;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.time.Instant;
 import java.util.List;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import org.bukkit.Skin;
+import tc.oc.pgm.util.translation.Translation;
 
 public class WebUtils {
 
-  private static final String RANDOM_NAME_API = "https://api.gamertag.dev/random"; // :)
-  private static final String USERNAME_API = "https://api.ashcon.app/mojang/v2/user/";;
+  // A big thanks to @Electroid for all these awesome APIs :)
+  private static final String RANDOM_NAME_API = "https://api.gamertag.dev/random";
+  private static final String USERNAME_API = "https://api.ashcon.app/mojang/v2/user/";
+  private static final String TRANSLATE_API = "https://api.gamertag.dev/translate";
+
+  public static CompletableFuture<Translation> getTranslated(
```

Comment:
1. Let's use a `Cache` for the last 100 messages.
2. Let's only send `Accept-Language` for locales that are online.